src/OrleansRuntime/GrainTypeManager/TypeManager.cs

```diff
@@ -1,33 +1,34 @@
 using System;
 using System.Threading.Tasks;
-using Orleans.Runtime.Providers;
+using Orleans.Streams;
 
 namespace Orleans.Runtime
 {
     internal class TypeManager : SystemTarget, ITypeManager
     {
         private readonly GrainTypeManager grainTypeManager;
+        private readonly ImplicitStreamSubscriberTable implicitStreamSubscriberTable;
 
-        internal TypeManager(SiloAddress myAddr, GrainTypeManager grainTypeManager)
+        internal TypeManager(
+            SiloAddress myAddr,
+            GrainTypeManager grainTypeManager,
+            ImplicitStreamSubscriberTable implicitStreamSubscriberTable)
             : base(Constants.TypeManagerId, myAddr)
         {
+            if (implicitStreamSubscriberTable == null) throw new ArgumentNullException(nameof(implicitStreamSubscriberTable));
```

Comment:
A minor thing, but this moved throwing of an ANE to the constructor `TypeManager` while it used to happen only when `GetImplicitStreamSubscriberTable` was called. Today this won't break because the table happens to get initialized inside `StreamingInitialize()` *before* `TypeManager` is instantiated. But this may easily break in the future if we rearrange things or make some of the streaming components pluggable and maybe even optional.